### PR TITLE
Add deepsped-autoTP-Fastapi serving

### PR DESCRIPTION
**Pull request by Romanticoseu**
## Description

Add deepsped-autoTP-Fastapi serving

### 1. Why the change?

Refer to this https://github.com/intel-analytics/ipex-llm/issues/10690

### 2. User API changes

None

### 4. How to test?
- [x] Application test

### CODE REVIEW

**Comment by glorysdj** [comment]
please add a readme

File: python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+from transformers import LlamaTokenizer, AutoTokenizer
+from deepspeed.accelerator.cpu_accelerator import CPU_Accelerator
+from deepspeed.accelerator import set_accelerator, get_accelerator
+from intel_extension_for_deepspeed import XPU_Accelerator
+import torch.distributed as dist
+
+model_path = "/mnt/disk1/models/Llama-2-7b-chat-hf"
+
+low_bit = "sym_int4"
+
```
Comment (glorysdj) [inline python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/serving.py:40]: try to make the model_path and low_bit can be passed as arguments

**Review by plusbang — COMMENTED** [review]
Please also update https://github.com/intel-analytics/ipex-llm/blob/main/python/llm/example/GPU/README.md#ipex-llm-examples-on-intel-gpu

File: python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+### 2. Run tensor parallel inference on multiple GPUs
+
+When we run the model in a distributed manner across two GPUs, the memory consumption of each GPU is only half of what it was originally, and the GPUs can work simultaneously during inference computation.
+
+We provide example usage for `Llama-2-7b` model running on Arc A770
+
+Run Llama-2-7b on two Intel Arc A770:
+
+```bash
+bash run_llama2_7b_arc_2_card.sh
```
Comment (glorysdj) [inline python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/README.md:38]: user need to modify this script to run this example

File: python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/run_llama2_7b_arc_2_card.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+export MASTER_ADDR=127.0.0.1
+export FI_PROVIDER=tcp
+export CCL_ATL_TRANSPORT=ofi
+export CCL_ZE_IPC_EXCHANGE=sockets
+export http_proxy=
```
[the remark this answers is not in the document]
Comment (Romanticoseu) [reply python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/run_llama2_7b_arc_2_card.sh:21]: I change this to `export no_proxy=localhost` to ensure the `curl` can correctly send request.

File: python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/README.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+
+We provide example usage for `Llama-2-7b-chat-hf` model running on Arc A770
+
+Run Llama-2-7b-chat-hf on two Intel Arc A770:
+
+```bash
+
+# Before run this script, you should adjust the YOUR_REPO_ID_OR_MODEL_PATH in last line
+# If you want to change server port, you can set port parameter in last line
+bash run_llama2_7b_chat_hf_arc_2_card.sh
```
Comment (plusbang) [inline python/llm/example/GPU/Deepspeed-AutoTP-FastAPI/README.md:41]: Different with the name of `.sh` script.

**Review by plusbang — COMMENTED** [review]
Other LGTM.

**Review by glorysdj — APPROVED** [review]
LGTM

**Pull request merged**